### ISSUE TITLE
Failure pulling submodules for recent version

### ISSUE THREAD

**Issue by de0a100** [OpenSpace/OpenSpace#2532]
When pulling the latest version of the master branch, it seems there are references to unpublished commits in submodules

fatal: remote error: upload-pack: not our ref 7a8bc211fc00f96f0636eceb2dc8ca91b331bc2d
fatal: the remote end hung up unexpectedly
fatal: Fetched in submodule path 'apps/OpenSpace/ext/sgct/ext/tracy', but it did not contain 7a8bc211fc00f96f0636eceb2dc8ca91b331bc2d. Direct fetching of that commit failed.
fatal: 
fatal: remote error: upload-pack: not our ref 7a8bc211fc00f96f0636eceb2dc8ca91b331bc2d
fatal: the remote end hung up unexpectedly
fatal: Fetched in submodule path 'ext/ghoul/ext/tracy', but it did not contain 7a8bc211fc00f96f0636eceb2dc8ca91b331bc2d. Direct fetching of that commit failed.
fatal: 
fatal: Failed to recurse into submodule path 'apps/OpenSpace/ext/sgct'
fatal: Failed to recurse into submodule path 'ext/ghoul'

**Comment by alexanderbock** [comment]
Huh, that is strange!  Could you verify what `origin` it uses on your end?  It ought to be `https://github.com/OpenSpace/tracy`  for both `ext/ghoul` (https://github.com/OpenSpace/Ghoul/blob/master/.gitmodules#L27)  and `apps/OpenSpace/ext/sgct`  (https://github.com/sgct/sgct/blob/master/.gitmodules#L15) and in that repository 7a8bc211fc exists:  https://github.com/OpenSpace/tracy/commit/7a8bc211fc00f96f0636eceb2dc8ca91b331bc2d

**Comment by de0a100** [comment]
Both  .gitmodules files are correct but, when I went to the submodule paths, I could see a different origin was used one the submodule was originally created:

~/OpenSpace/apps/OpenSpace/ext/sgct/ext/tracy$ git remote -v
origin  https://bitbucket.org/wolfpld/tracy.git (fetch)
origin  https://bitbucket.org/wolfpld/tracy.git (push)

When I manually set "tracy" origin to the URL you provided, the problem was solved!
This must have been a local problem on my side, probably updating from an old version. Thanks for your help!
